From 5691f2b93484dd9cebae5a9ca61e9de44e814d15 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?A=20U=20Thor?= <thor@redhat.com>
Date: Wed, 8 Sep 2021 09:45:13 +0200
Subject: [PATCH] Funky
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 hello.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hello.rs b/hello.rs
index f6a5a44..d1ccfb6 100644
--- a/hello.rs
+++ b/hello.rs
@@ -1,3 +1,4 @@
 fn main() {
     println!("Hello Fedora Linux!");
+    println!("Hello World!");
 }
-- 
2.31.1

